scripts/asr_language_modeling/ngram_lm/eval_wfst_decoding_ctc.py

```diff
@@ -0,0 +1,441 @@
+        punctuation_marks = ".,?",
+        separate_punctuation = False,
+        do_lowercase = False,
+        rm_punctuation = False,
+    ))
+# fmt: on
+
+
+def beam_search_eval(
+    model: nemo_asr.models.ASRModel,
+    cfg: EvalWFSTNGramConfig,
+    all_probs: List[torch.Tensor],
+    target_transcripts: List[str],
+    preds_output_file: str = None,
+    lm_weight: float = 1.0,
+    beam_width: float = 10.0,
+    beam_batch_size: int = 512,
+    progress_bar: bool = True,
+    punctuation_capitalization: PunctuationCapitalization = None,
+):
+    level = logging.getEffectiveLevel()
+    logging.setLevel(logging.CRITICAL)
+    # Reset config
+    if isinstance(model, EncDecHybridRNNTCTCModel):
+        model.change_decoding_strategy(decoding_cfg=None, decoder_type="ctc")
+    else:
+        model.change_decoding_strategy(None)
+
+    # Override the beam search config with current search candidate configuration
+    cfg.decoding.beam_width = beam_width
+    cfg.decoding.lm_weight = lm_weight
+    cfg.decoding.open_vocabulary_decoding = cfg.open_vocabulary_decoding
+    cfg.decoding.return_best_hypothesis = False
+    cfg.decoding.arpa_lm_path = cfg.arpa_model_file
+    cfg.decoding.wfst_lm_path = cfg.decoding_wfst_file
+    cfg.decoding.device = cfg.device
+    cfg.decoding.decoding_mode = cfg.decoding_mode
+    cfg.decoding.search_type = cfg.decoding_search_type
+
+    # Update model's decoding strategy config
+    model.cfg.decoding.strategy = "wfst"
+    model.cfg.decoding.wfst = cfg.decoding
+
+    # Update model's decoding strategy
+    if isinstance(model, EncDecHybridRNNTCTCModel):
+        model.change_decoding_strategy(model.cfg.decoding, decoder_type='ctc')
+        decoding = model.ctc_decoding
+    else:
+        model.change_decoding_strategy(model.cfg.decoding)
+        decoding = model.decoding
+    logging.setLevel(level)
+
+    wer_dist_first = cer_dist_first = 0
+    wer_dist_best = cer_dist_best = 0
+    words_count = 0
+    chars_count = 0
+    sample_idx = 0
+    if preds_output_file:
+        out_file = open(preds_output_file, 'w', encoding='utf_8', newline='\n')
+
+    if progress_bar:
+        it = tqdm(
+            range(int(np.ceil(len(all_probs) / beam_batch_size))),
+            desc=f"Beam search decoding with width={beam_width}, lm_weight={lm_weight}",
+            ncols=120,
+        )
+    else:
+        it = range(int(np.ceil(len(all_probs) / beam_batch_size)))
+    for batch_idx in it:
+        # disabling type checking
+        probs_batch = all_probs[batch_idx * beam_batch_size : (batch_idx + 1) * beam_batch_size]
+        probs_lens = torch.tensor([prob.shape[0] for prob in probs_batch])
+        with torch.no_grad():
+            packed_batch = torch.zeros(len(probs_batch), max(probs_lens), probs_batch[0].shape[-1], device='cpu')
+
+            for prob_index in range(len(probs_batch)):
+                packed_batch[prob_index, : probs_lens[prob_index], :] = torch.tensor(
```

Comment:
use `probs_batch[prob_index].to(device=...)`

See warning from PyTorch:
```
UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
```